Components/Model/Category.php

```diff
@@ -122,4 +123,17 @@ public function setCategoryPath($categoryPath)
 
         return $this;
     }
+
+    /**
+     * Returns the HTML to render categories
+     *
+     * @return MarkupableString
+     */
+    public function getAbstractObject()
```

Comment:
Instead of introducing a method I think you could extend the `MarkupableString` directly. Similar implimentation what we have here: https://github.com/Nosto/nosto-prestashop/blob/develop/classes/models/NostoCategory.php